synapse/axon.py

```diff
@@ -992,22 +1028,100 @@ async def iterMpkFile(self, sha256):
             for _, item in unpk.feed(byts):
                 yield item
 
+    @contextlib.asynccontextmanager
+    async def open(self, sha256, mode='r'):
+
+        link, fobj = await s_link.linkfile(mode=mode)
+
+        async def fill():
+            async for byts in self.get(sha256):
+                await link.send(byts)
+            await link.fini()
+
+        self.schedCoro(fill())
+
+        try:
+            yield fobj
+
+        finally:
+            fobj.close()
+            await link.fini()
+
+    def _readlines(self, fobj, queue, sha256):
+            try:
+                for line in fobj:
+                    line = line.rstrip('\n')
+                    queue.put(line)
```

Comment:
Does this ever block or provide back pressure? I think we may risk filling up RAM here?